src/printer.js

```diff
@@ -2732,14 +2732,13 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
       path.call(print, "right")
     ]);
 
-    // If there's only a single binary expression: everything except && and ||,
-    // we want to create a group in order to avoid having a small right part
-    // like -1 be on its own line.
+    // If there's only a single binary expression, we want to create a group
+    // in order to avoid having a small right part like -1 be on its own line.
     const parent = path.getParentNode();
-    const shouldGroup = node.type === "BinaryExpression" &&
-      parent.type !== "BinaryExpression" &&
-      node.left.type !== "BinaryExpression" &&
-      node.right.type !== "BinaryExpression";
+    const shouldGroup =
+      parent.type !== node.type &&
+      node.left.type !== node.type &&
+      node.right.type !== node.type;
```

Comment:
If you are in this code path, this means that node.type is already Logical OR Binary. This is the code that handles those cases.

Before we had node.type is Binary because I didn't want to make that behavior for Logical as I wasn't sure what the consequences would be. But now I do want to do the same for both, so I can just get rid of the check altogether.